tests/factories/__init__.py

```diff
@@ -27,6 +28,9 @@ def set_sqlalchemy_session(session):
         # pylint:disable=protected-access
         factory_class._meta.sqlalchemy_session = session
 
+        if persistence:
+            factory_class._meta.sqlalchemy_session_persistence = persistence
```

Comment:
The functional tests need Factory Boy to commit the DB session after creating each model instance, so they do `set_sqlalchemy_session(session, persistence="commit")`